lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1671,13 +1671,7 @@ def plot_surface(self, X, Y, Z, *args, **kwargs):
             color = np.array(mcolors.to_rgba(color))
             fcolors = None
 
-        cmap = kwargs.get('cmap', None)
-        norm = kwargs.pop('norm', None)
-        vmin = kwargs.pop('vmin', None)
-        vmax = kwargs.pop('vmax', None)
-        linewidth = kwargs.get('linewidth', None)
         shade = kwargs.pop('shade', cmap is None)
```

Comment:
*shade* could be converted as well (you already did it for plot_trisurf :smile:)

Reply:
An explicit `shade=None` was interpreted as `shade=False`. I suppose this is also an unintended implementation detail. I'm happy to discuss this separately.

Reply:
See #11154.